tests/test_templates.py

```diff
@@ -37,282 +47,308 @@
+                                 [[0.93, 1.58], [5.07, 4.82]],
+                                 [[0.21,  0.12], [-0.09, 0.04]],
+                                 [[4.76, 6.08], [6.09, 6.22]],
+                                 [[4.83], [1.70]],
+                                 [[4.74], [5.39]],
+                                 [[0.88, 0.62], [1.09, 3.02]],
+                                 [[-0.01, -0.05], [0.08, -0.19]],
+                                 [[1.89, 3.59], [1.49, 3.71]],
+                                 [[0.09,  0.03], [-0.14,  0.04]]
                                  ]),
-            (Interferometer, [[2.33312851], [3.49488327], [0.9868003, 1.58798724]])
+            (Interferometer, [[2.31], [3.49], [0.98, 1.54]])
+            ]
+
+# qubit templates, constant inputs, and ``argnum`` argument of qml.grad
+qubit_grad = [(StronglyEntanglingLayers, [[[[4.54, 4.79, 2.98], [4.93, 4.11, 5.58]],
+                                           [[6.08, 5.94, 0.05], [2.44, 5.07, 0.95]]]], [0]),
+               (RandomLayers, [[[0.56, 5.14], [2.21, 4.27]]], [0]),
+               (AngleEmbedding, [[1., 2.]], [0])
+              ]
+
+# cv templates, constant inputs, and ``argnum`` argument of qml.grad
+cv_grad = [(DisplacementEmbedding, [[1., 2.]], [0]),
+            (SqueezingEmbedding, [[1., 2.]], [0]),
+            (CVNeuralNetLayers, [[[2.31], [1.22]],
+                                 [[3.47], [2.01]],
+                                 [[0.93, 1.58], [5.07, 4.82]],
+                                 [[0.21,  0.12], [-0.09, 0.04]],
+                                 [[4.76, 6.08], [6.09, 6.22]],
+                                 [[4.83], [1.70]],
+                                 [[4.74], [5.39]],
+                                 [[0.88, 0.62], [1.09, 3.02]],
+                                 [[-0.01, -0.05], [0.08, -0.19]],
+                                 [[1.89, 3.59], [1.49, 3.71]],
+                                 [[0.09,  0.03], [-0.14,  0.04]]
+                                 ], list(range(11))),
+            (Interferometer, [[2.31], [3.49], [0.98, 1.54]], [0, 1, 2])
             ]
 
+#########################################
+# Circuits
+
+
+def qnode_qubit_args(dev, intrfc, templ1, templ2, n):
+    """QNode for qubit integration circuit using positional arguments"""
+    @qml.qnode(dev, interface=intrfc)
+    def circuit(*inp_):
+        # Split inputs again
+        inp1_ = inp_[:n]
+        inp2_ = inp_[n:]
+
+        qml.PauliX(wires=0)
+        templ1(*inp1_, wires=range(2))
+        templ2(*inp2_, wires=range(2))
+        qml.PauliX(wires=1)
+        return [qml.expval(qml.Identity(0)), qml.expval(qml.PauliX(1))]
+    return circuit
+
+
+def qnode_qubit_kwargs(dev, intrfc, templ1, templ2, n):
+    """QNode for qubit integration circuit using keyword arguments"""
+    @qml.qnode(dev, interface=intrfc)
+    def circuit(**inp_):
+        # Split inputs again
+        ks = [int(k) for k in inp_.keys()]
```

Comment:
Is `ks` here just to ensure a fixed ordering?

Reply:
Yes, I did not trust string ordering with potentially large numbers...